<a href="https://colab.research.google.com/github/arechigajd/machinelearningprojectII/blob/main/Machine_Learning_Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd

In [3]:
filename = ('/content/water_potability.csv')
watpot = pd.read_csv(filename)
watpot.head()

ph    Hardness        Solids  Chloramines     Sulfate  Conductivity  \
0       NaN  204.890455  20791.318981     7.300212  368.516441    564.308654   
1  3.716080  129.422921  18630.057858     6.635246         NaN    592.885359   
2  8.099124  224.236259  19909.541732     9.275884         NaN    418.606213   
3  8.316766  214.373394  22018.417441     8.059332  356.886136    363.266516   
4  9.092223  181.101509  17978.986339     6.546600  310.135738    398.410813   

   Organic_carbon  Trihalomethanes  Turbidity  Potability  
0       10.379783        86.990970   2.963135           0  
1       15.180013        56.329076   4.500656           0  
2       16.868637        66.420093   3.055934           0  
3       18.436524       100.341674   4.628771           0  
4       11.558279        31.997993   4.075075           0

In [5]:
watpot.shape

(3276, 10)

1. The source of this dataset is Kaggle
2. The dataset is about water potability, use and consumption
3. The target is whether water is suitable for consumption or not
4. This is a classification problem
5. 10 features
6. 3276 rows of data
7. Challenges I foresee in this data would be to find the best features and preprocess the data correctly with no data leakage.

In [4]:
path = ('/content/diabetes.csv')
diab = pd.read_csv(path)
diab.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [7]:
diab.shape

(768, 9)

1. Source of the data set is Kaggle
2. The data is about women Pima Indians and their possibility of getting diabetes
3. The target is the outcome, whether or not they will get diabetes.
4. This is a classification problem
5. 8 possible features
6. 768 rows of data
7. Challenge I foresee is modeling the data to ensure a minimal amount of false negatives.